File: src/main/java/com/hsmchurch/app/noticeboard/domain/NoticeRepositoryCustom.java
package com.hsmchurch.app.noticeboard.domain;

import com.hsmchurch.app.noticeboard.ui.response.NoticeResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface NoticeRepositoryCustom {

    Page<NoticeResponse> findListAll(Pageable pageable);

    NoticeResponse findOneWithWriterName(Long noticeId);
}

File: src/main/java/com/hsmchurch/app/noticeboard/domain/NoticeRepositoryCustom.kt
package com.hsmchurch.app.noticeboard.domain

import com.hsmchurch.app.noticeboard.ui.response.NoticeResponse
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface NoticeRepositoryCustom {

    fun findListAll(pageable: Pageable): Page<NoticeResponse?>

    fun findOneWithWriterName(noticeId: Long): NoticeResponse?
}
